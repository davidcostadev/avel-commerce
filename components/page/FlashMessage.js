import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

export const FlashMessageItem = ({ type, message }) => (
  <div className={classnames(
    'alert',
    {
      'alert-success': type === 'success',
      'alert-danger': type === 'danger',
      'alert-warning': type === 'warning',
      'alert-info': type === 'info',
    },
  )}
  >
    {message}
  </div>
)

FlashMessageItem.propTypes = {
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
}

export const FlashMessages = ({ msgs }) => {
  if (!msgs.length) {
    return null
  }

  return msgs.map(({ id, type, message }) => (
    <FlashMessageItem
      key={id}
      type={type}
      message={message}
    />
  ))
}

FlashMessages.propsTypes = {
  msgs: PropTypes.shape({
    id: PropTypes.any.isRequired,
    type: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
  }).isRequired,
}

