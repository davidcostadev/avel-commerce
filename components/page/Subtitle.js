import React from 'react'
import {
  DescriptionLogo,
} from '../../layout/Header'

const SubTitle = () => {
  if (!process.env.BUSSNESS_LOGO_DESCRIPTION.length) {
    return null
  }

  return (
    <DescriptionLogo>{process.env.BUSSNESS_LOGO_DESCRIPTION}</DescriptionLogo>
  )
}

export default SubTitle
