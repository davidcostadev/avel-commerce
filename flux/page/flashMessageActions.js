import { ADD_FLASH_MESSAGE } from '../type'

export const addFlashMessage = message => ({
  type: ADD_FLASH_MESSAGE,
  payload: message,
})

export default addFlashMessage
