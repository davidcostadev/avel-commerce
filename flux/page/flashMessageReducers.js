import shortid from 'shortid'
import { ADD_FLASH_MESSAGE } from '../type'

export default (state = [], { type, payload }) => {
  switch (type) {
    case ADD_FLASH_MESSAGE:
      return [
        ...state,
        {
          id: shortid.generate(),
          type: payload.type,
          message: payload.message,
        },
      ]
    default: return state
  }
}
