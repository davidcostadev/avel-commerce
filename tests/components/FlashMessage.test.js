/* eslint no-undef: "off" */
import React from 'react'
import renderer from 'react-test-renderer'
import { FlashMessages } from '../../components/page/FlashMessage'

const mockErrors = [
  {
    id: 1,
    type: 'success',
    message: 'message 01',
  },
  {
    id: 2,
    type: 'danger',
    message: 'message 02',
  },
  {
    id: 3,
    type: 'info',
    message: 'message 03',
  },
  {
    id: 4,
    type: 'warning',
    message: 'message 04',
  },
]

describe('Component <FlashMessages />', () => {
  it('With list erros', () => {
    const tree = renderer.create((
      <FlashMessages
        msgs={mockErrors}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('With no list erros', () => {
    const tree = renderer.create((
      <FlashMessages
        msgs={[]}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
