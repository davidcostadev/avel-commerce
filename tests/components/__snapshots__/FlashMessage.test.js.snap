// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FlashMessages /> With list erros 1`] = `
Array [
  <div
    className="alert alert-success"
  >
    message 01
  </div>,
  <div
    className="alert alert-danger"
  >
    message 02
  </div>,
  <div
    className="alert alert-info"
  >
    message 03
  </div>,
  <div
    className="alert alert-warning"
  >
    message 04
  </div>,
]
`;

exports[`Component <FlashMessages /> With no list erros 1`] = `null`;
